pkg/build/graph/analysis/bc.go

```diff
@@ -98,3 +102,87 @@ func FindCircularBuilds(g osgraph.Graph) []osgraph.Marker {
 
 	return markers
 }
+
+// FindPendingTags inspects all imageStreamTags that serve as outputs to builds.
+//
+// Precedence of failures:
+// 1. A build config points to the non existent tag but no current build exists.
+// 2. A build config points to the non existent tag but the latest build has failed.
+func FindPendingTags(g osgraph.Graph) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+ist:
+	for _, uncastIstNode := range g.NodesByKind(imagegraph.ImageStreamTagNodeKind) {
+		istNode := uncastIstNode.(*imagegraph.ImageStreamTagNode)
+		if bcNode, points := buildPointsToTag(g, uncastIstNode); points && !istNode.Found() {
+			buildNode := latestBuild(g, bcNode)
+
+			// A build config points to the non existent tag but no current build exists.
+			if buildNode == nil {
+				markers = append(markers, osgraph.Marker{
+					Node:         bcNode,
+					RelatedNodes: []graph.Node{uncastIstNode},
+
+					Severity:   osgraph.WarningSeverity,
+					Key:        TagNotAvailableWarning,
+					Message:    fmt.Sprintf("%s needs to be imported or created by a build.", istNode.ResourceString()),
+					Suggestion: osgraph.Suggestion(fmt.Sprintf("oc start-build %s", bcNode.(*buildgraph.BuildConfigNode).ResourceString())),
+				})
+				continue ist
+			}
+
+			// A build config points to the non existent tag but something is going on with
+			// the latest build.
+			// TODO: Handle other build phases.
+			switch buildNode.(*buildgraph.BuildNode).Build.Status.Phase {
+			case buildapi.BuildPhaseCancelled:
+				// TODO: Add a warning here.
+			case buildapi.BuildPhaseError:
+				// TODO: Add a warning here.
+			case buildapi.BuildPhaseComplete:
+				// We should never hit this. The output of our build is missing but the build is complete.
+				// Most probably the user has messed up?
+				// TODO: What should we do here?
+			case buildapi.BuildPhaseFailed:
+				// Since the tag hasn't been populated yet, we assume there hasn't been a successful
+				// build so far.
+				markers = append(markers, osgraph.Marker{
+					Node:         buildNode,
+					RelatedNodes: []graph.Node{uncastIstNode, bcNode, buildNode},
+
+					Severity:   osgraph.ErrorSeverity,
+					Key:        LatestBuildFailedErr,
+					Message:    fmt.Sprintf("%s has failed.", buildNode.(*buildgraph.BuildNode).ResourceString()),
+					Suggestion: osgraph.Suggestion(fmt.Sprintf("Inspect the build failure with 'oc logs %s'", buildNode.(*buildgraph.BuildNode).ResourceString())),
+				})
+				continue ist
+			default:
+				// Do nothing when latest build is new, pending, or running.
+			}
+		}
+	}
+
+	return markers
+}
+
+func buildPointsToTag(g osgraph.Graph, istag graph.Node) (graph.Node, bool) {
+	for _, bcNode := range g.PredecessorNodesByEdgeKind(istag, buildedges.BuildOutputEdgeKind) {
+		return bcNode, true
+	}
+	return nil, false
+}
+
+func latestBuild(g osgraph.Graph, bc graph.Node) graph.Node {
```

Comment:
> Why not return the exact kind of node you expect?

Given the guarantees of the graph, I'm ok with a hard cast that panics if the graph was built wrong.